scipy/integrate/_rules/_base.py

```diff
@@ -0,0 +1,657 @@
+        cubature rules (quadratures) and more generally for N-D cubature rules, it
+        should be of shape ``(num_nodes, ndim)``. ``weights`` should be of shape
+        `(num_nodes,)`. The nodes and weights should be for integrals over `[-1, 1]^n`.
+
+    See Also
+    --------
+    NestedRule, NewtonCotesQuad, GaussLegendreQuad, GaussKronrodQuad, GenzMalikCub
+
+    Examples
+    --------
+
+    Implementing Simpson's 1/3 rule:
+
+    >>> import numpy as np
+    >>> from scipy.integrate._rules import FixedRule
+    >>> class SimpsonsQuad(FixedRule):
+    ...     @property
+    ...     def nodes_and_weights(self):
+    ...         nodes = np.array([-1, 0, 1])
+    ...         weights = np.array([1/3, 4/3, 1/3])
+    ...         return (nodes, weights)
+    >>> rule = SimpsonsQuad()
+    >>> rule.estimate(
+    ...     f=lambda x: x**2,
+    ...     a=np.array([0]),
+    ...     b=np.array([1]),
+    ... )
+     [0.3333333]
+    """
+
+    @property
+    def nodes_and_weights(self):
+        raise NotImplementedError
+
+    def estimate(self, f, a, b, args=(), kwargs=None):
+        r"""
+        Calculate estimate of integral of ``f`` in rectangular region described by
+        corners ``a`` and ``b`` as ``sum(weights * f(nodes))``.
+
+        Nodes and weights will automatically be adjusted from calculating integrals over
+        :math:`[-1, 1]^n` to :math:`[a, b]^n`.
+
+        Parameters
+        ----------
+        f : callable
+            Function to integrate. ``f`` must have the signature::
+                f(x : ndarray, \*args, \*\*kwargs) -> ndarray
+
+            ``f`` should accepts arrays ``x`` of shape::
+                (npoints, ndim)
+
+            and output arrays of shape::
+                (npoints, output_dim_1, ..., output_dim_n)
+
+            In this case, ``estimate`` will return arrays of shape::
+                (output_dim_1, ..., output_dim_n)
+        a, b : ndarray
+            Lower and upper limits of integration as rank-1 arrays specifying the left
+            and right endpoints of the intervals being integrated over. Infinite limits
+            are currently not supported.
+        args : tuple, optional
+            Additional positional args passed to ``f``, if any.
+        kwargs : tuple, optional
+            Additional keyword args passed to ``f``, if any.
+
+        Returns
+        -------
+        est : ndarray
+            Result of estimation. If ``f`` returns arrays of shape ``(npoints,
+            output_dim_1, ..., output_dim_n)``, then ``est`` will be of shape
+            ``(output_dim_1, ..., output_dim_n)``.
+        """
+        nodes, weights = self.nodes_and_weights
+
+        return _apply_fixed_rule(f, a, b, nodes, weights, args, kwargs)
+
+
+class NestedRule(Rule):
+    r"""
+    A rule with error estimate given by the difference between two underlying rules. If
+    constructed as ``NestedRule(higher, lower)``, this will use::
```

Comment:
```suggestion
    r"""
    A rule with error estimate given by the difference between two underlying rules.
    
    If constructed as ``NestedRule(higher, lower)``, this will use::
```

and a few others